piprot/piprot.py

```diff
@@ -364,90 +390,33 @@ def main(
-
-    cli_parser.add_argument(
-        '-b', '--branch',
-        help='The branch to test requirements from, used with '
-             'the Github URL support.')
-
-    cli_parser.add_argument(
-        '-t', '--token',
-        help='Github personal access token to be used with '
-             'the Github URL support.')
-
-    cli_parser.add_argument(
-        '-p', '--path',
-        help='Path to requirements file in remote repository.')
-
-    cli_parser.add_argument(
-        '-d', '--delay',
-        help='Delay before an outdated package triggers an error.'
-             '(in days, default to 1).')
-
-    cli_parser.add_argument('-u', '--url',
-                            help='URL to requirements file.')
-
-    # if there is a requirements.txt file, use it by default. Otherwise print
-    # usage if there are no arguments.
-    nargs = '+'
-
-    if (
-        '--github' in sys.argv
-        or '-g' in sys.argv
-        or '-u' in sys.argv
-        or '--url' in sys.argv
-    ):
-        nargs = "*"
-
-    default = None
-    if os.path.isfile('requirements.txt'):
-        nargs = "*"
-        default = [open('requirements.txt')]
-
-    cli_parser.add_argument('file', nargs=nargs, type=argparse.FileType(),
-                            default=default, help='requirements file(s), use '
-                                                  '`-` for stdin')
-
-    cli_args = cli_parser.parse_args()
-
-    if len(cli_args.file) > 1 and cli_args.verbatim:
-        sys.exit('--verbatim only allowed for single requirements files')
+    options = docopt(__doc__, version=__version__)
+
+    if options['<file>'] and options['--verbatim']:
+        sys.exit('--verbatim only allowed for single requirements files.')
 
     verbose = True
-    if cli_args.quiet:
+    if options['--quiet']:
         verbose = False
-    elif cli_args.verbatim:
+    elif options['--verbatim']:
         verbose = False
 
+    req_files = []
+    for filename in options['<file>']:
+        if options['<file>'] == ['-']:
+            req_files.append(sys.stdin)
+        else:
+            req_files.append(open(filename))
+    else:
+        if not options['--github'] and not options['--url']:
+            # No source provided, read from stdin.
+            req_files.append(sys.stdin)
+
     # call the main function to kick off the real work
-    main(req_files=cli_args.file, verbose=verbose, outdated=cli_args.outdated,
-         latest=cli_args.latest, verbatim=cli_args.verbatim,
-         repo=cli_args.github, branch=cli_args.branch, path=cli_args.path,
-         token=cli_args.token, url=cli_args.url, delay=cli_args.delay)
+    main(req_files=options['<file>'], verbose=verbose, outdated=options['--outdated'],
+         latest=options['--latest'], verbatim=options['--verbatim'],
+         repo=options['--github'], branch=options['--branch'], path=options['--path'],
+         token=options['--token'], url=options['--url'], delay=options['--delay'])
```

Comment:
Didn't you mean to use the `req_files` list you constructed above?

Reply:
Indeed.